### ISSUE TITLE
Failure to resolve the DNS name of the statsd server brings locker down hard

### ISSUE THREAD

**Issue by mdz** [LockerProject/Locker#916]
```
statsd error: Error: getaddrinfo ENOENT
10 Mar 02:32:14 - error: Uncaught exception:
10 Mar 02:32:14 - error: { [Error: getaddrinfo ENOENT] code: 'ENOTFOUND', errno: 'ENOTFOUND', syscall: 'getaddrinfo' }
10 Mar 02:32:14 - error: 'Error: getaddrinfo ENOENT\n    at errnoException (dns.js:31:11)\n    at Object.onanswer [as oncomplete] (dns.js:140:16)'

10 Mar 02:32:14 - info: Shutting down...
statsd error: Error: getaddrinfo ENOENT
10 Mar 02:32:14 - error: Uncaught exception:
10 Mar 02:32:14 - error: { [Error: getaddrinfo ENOENT] code: 'ENOTFOUND', errno: 'ENOTFOUND', syscall: 'getaddrinfo' }
10 Mar 02:32:14 - error: 'Error: getaddrinfo ENOENT\n    at errnoException (dns.js:31:11)\n    at Object.onanswer [as oncomplete] (dns.js:140:16)'
Aieee, shutdown called while already shutting down!  Aborting!
locker was only running for 2 seconds, not respawning
```

**Comment by othiym23** [comment]
Derp. Asynchronous and synchronous error-handling can't mix. Had to revert my change. Will take this up with the Node team.

**Comment by mdz** [comment]
Is this still open, or was it addressed by d3cea88?

**Comment by othiym23** [comment]
It's closed. I'm poking at the Node team on the corresponding bug on node, because it seems weird to have the dgram code assume a listener stream, but I think the pattern the stats code has in place now is how Node wants things done today.
